pkg/controller/kieapp/shared/utils.go

```diff
@@ -174,3 +180,51 @@ func GetNamespacedName(object metav1.Object) types.NamespacedName {
 		Namespace: object.GetNamespace(),
 	}
 }
+
+func GetConsoleYAMLSample(res resource.KubernetesResource) (*consolev1.ConsoleYAMLSample, error) {
+	snippetStr := res.GetAnnotations()["consoleSnippet"]
+	var snippet bool = false
+	if snippetStr != "" {
+		tmp, err := strconv.ParseBool(snippetStr)
+		if err != nil {
+			return nil, errors.New("Unable to parse snippet to boolean.")
+		}
+		snippet = tmp
+	}
+
+	title, _ := res.GetAnnotations()["consoleTitle"]
+	desc, _ := res.GetAnnotations()["consoleDesc"]
+	name, _ := res.GetAnnotations()["consoleName"]
+	if name == "" {
+		name = res.GetName() + "-yamlsample"
+	}
+
+	curannos := res.GetAnnotations()
```

Comment:
How about giving this a better name (maybe just `annotations`?), moving this further up, and using it to retrieve the values too?

Reply:
modified